src/sql/src/parser.rs

```diff
@@ -175,6 +175,32 @@ impl<'a> ParserContext<'a> {
         }
     }
 
+    /// Parses MySQL style 'PREPARE stmt_name FROM stmt' into a (stmt_name, stmt) tuple.
+    pub fn parse_mysql_prepare_stmt(
+        sql: &'a str,
+        dialect: &dyn Dialect,
+    ) -> Result<(String, String)> {
+        let parser = Parser::new(dialect)
```

Comment:
Good job! I think we can make this function as the member function of `ParserContext`. Then we can use it like:

```rust
let mut parser_ctx = ParserContext::new(dialect, sql)?;

let _stmt  = parser_ctx.parse_mysql_prepare_stmt()?

```

The same suggestion as `parse_mysql_execute_stmt`.

Reply:
It seems a little tricky for me with some async issues. I'm not quite sure but here's my understanding: `ParserContext` takes a `Parser` as a member and `Parser` uses a `Rc<Cell<usize>>` as its `recursion_counter`, thus the struct is not `Send`, so we cannot use it inside the async block of the `on_query` function.

Reply:
Yes. That's what I mean.